src/main/java/com/redhat/devtools/intellij/quarkus/classpath/ClasspathResourceChangedManager.java

```diff
@@ -0,0 +1,109 @@
+import com.intellij.openapi.util.Pair;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.openapi.vfs.VirtualFileManager;
+import com.intellij.psi.PsiManager;
+import com.intellij.util.messages.MessageBusConnection;
+import com.intellij.util.messages.Topic;
+import com.intellij.workspaceModel.ide.WorkspaceModelTopics;
+import com.redhat.devtools.intellij.quarkus.classpath.bridge.WorkspaceModelTopicsBridge;
+
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Classpath resource change manager provides the capability to track update of libraries changed and Java, microprofile-config properties files
+ * by any component by registering a listener {@link Listener}.
+ *
+ * <code>
+ * ClasspathResourceChangeManager.Listener myListener = ...
+ * project.getMessageBus().connect(project).subscribe(ClasspathResourceChangeManager.TOPIC, myListener);
+ * </code>
+ *
+ *
+ * <ul>
+ *     <li>Track update of libraries is done with {@link WorkspaceModelTopics}.
+ *     In other words {@link Listener#librariesChanged()}  are fired when all libraries are inserted, deleted, updated.</li>
+ *     <li>Track update of Java, microprofile-config properties files are done when Java Psi file is updated, when Java file is created, deleted, saved.</li>
+ * </ul>
+ */
+public class ClasspathResourceChangedManager implements Disposable {
+
+    public static final Topic<ClasspathResourceChangedManager.Listener> TOPIC = Topic.create(ClasspathResourceChangedManager.class.getName(), ClasspathResourceChangedManager.Listener.class);
+
+    private SourceFileChangeNotifier sourcesChanged;
+    private MessageBusConnection connection;
+    private ClasspathSourceFileChangedListener listener;
+
+    public static ClasspathResourceChangedManager getInstance(Project project) {
+        return ServiceManager.getService(project, ClasspathResourceChangedManager.class);
+    }
+
+    public interface Listener {
+
+        void librariesChanged();
+
+        void sourceFilesChanged(Set<Pair<VirtualFile, Module>> sources);
+    }
+
+    private final Project project;
+
+    public ClasspathResourceChangedManager(Project project) {
+        this.project = project;
+    }
+
+    private synchronized void initialize(Project project) {
+        if (sourcesChanged != null) {
+            return;
+        }
+        // Send source files changed in debounce mode
+        this.sourcesChanged = new SourceFileChangeNotifier(files ->
+                project.getMessageBus().syncPublisher(ClasspathResourceChangedManager.TOPIC).sourceFilesChanged(files)
+        );
+        listener = new ClasspathSourceFileChangedListener(project, sourcesChanged);
+        connection = project.getMessageBus().connect();
+        // Track end of Java libraries update
+        WorkspaceModelTopicsBridge workspaceModelTopicsBridge = new WorkspaceModelTopicsBridge();
+        Object libraryChangedListener = new ClasspathLibraryChangedListener(project).getProxyInstance();

```

Comment:
no need to create a proxy instance if workspaceModelTopicsBridge.CHANGED == null